pkg/cmd/server/origin/asset.go

```diff
@@ -141,6 +142,18 @@ func (c *AssetConfig) buildAssetHandler() (http.Handler, error) {
 	return handler, nil
 }
 
+// Have to convert to arrays because go templates are limited and we need to be able to know
+// if we are on the last index for trailing commas in JSON
+func extensionPropertyArrays(extensionProperties map[string]string) ([]template.JSStr, []template.JSStr) {
+	extensionKeys := []template.JSStr{}
```

Comment:
I agree it seems like something in the YAML parser was misbehaving but not
sure how to fix that?

html/template JSStr is not the same thing as the js filter in text/template
 they have different escaping behaviors, i wasn't running JSStr and then
still running the js filter, i was only using JSStr.  Its definitely not
ideal because of the need to double escape the backslash on newline chars.
But that seemed like the lesser of two evils when comparing what using the
js filter was doing to `\"`

On Fri, May 20, 2016 at 11:23 AM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/cmd/server/origin/asset.go
> <https://github.com/openshift/origin/pull/8934#discussion_r64059134>:
>
> > @@ -141,6 +142,18 @@ func (c *AssetConfig) buildAssetHandler() (http.Handler, error) {
> >  	return handler, nil
> >  }
> >
> > +// Have to convert to arrays because go templates are limited and we need to be able to know
> > +// if we are on the last index for trailing commas in JSON
> > +func extensionPropertyArrays(extensionProperties map[string]string) ([]template.JSStr, []template.JSStr) {
> > +	extensionKeys := []template.JSStr{}
>
> we really shouldn't need to pre-escape these... the template should be
> able to do that for us. I suspect we had weird values in the strings
> (possibly from YAML doing odd things with quotes within quotes?)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8934/files/21217f567be4714edada6e6301ec46d2e6ff706b#r64059134>
>